src/garage/experiment/experiment.py

```diff
@@ -234,6 +235,43 @@ def _make_sequential_log_dir(log_dir):
             i += 1
 
 
+def _make_experiment_signature(function):
+    """Generate an ExperimentTemplate's signature from its function.
+
+    Checks that the first parameter is named ctxt and removes it from the
+    signature. Makes all other parameters keyword only.
+
+    Args:
+        function (callable[ExperimentContext, ...]): The wrapped function.
+
+    Returns:
+        inspect.Signature: The signature of the ExperimentTemplate.
+
+    Raises:
+        ValueError: If the wrapped function's first parameter is not 'ctxt'.
+
+    """
+    func_sig = inspect.signature(function)
+    new_params = []
+    is_first_param = True
+    for param in func_sig.parameters.values():
+        if is_first_param:
+            # Don't output it to the experiment params, since it will contain
+            # the context.
+            if param.name != 'ctxt':
+                raise ValueError('Experiment functions should have a first '
```

Comment:
Also need to check if `function` has an empty parameter list.

Reply:
Oh! Good catch.